File: Python/Parite2/Exo4.py
import os

numero = [0]  # variable globale simulée

for i in range(1, 6):
    numero[0] = i
    pid = os.fork()
    if pid == 0:
        print(f"je suis le fils {numero[0]}")
        os._exit(0)
for _ in range(5):
    os.wait()

File: C/Parite2/Exo4.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>
#include<time.h>

int main()
{
    int nb_fils = 5;
    pid_t pid;

    for (int i = 0; i < nb_fils; i++)
    {
        pid = fork();
        if (pid == 0)
        {
            printf("Je suis le fils %d\n", i + 1);
            break;
        }
    }
    return (0);
}